# Embeddings demo

In [3]:
from langchain_google_genai import GoogleGenerativeAIEmbeddings
embeddings_model = GoogleGenerativeAIEmbeddings(model="models/embedding-001")

In [4]:
embeddings = embeddings_model.embed_documents(
    [
        "Hi there!",
        "Oh, hello!",
        "What's your name?",
        "My friends call me World",
        "Hello World!"
    ]
)
len(embeddings), len(embeddings[0])

(5, 768)

* The length of embedding vectors is 768 (It remains the same for all sentences).

## How is dermatology different from neurology? :)

In [5]:
embedded_query = embeddings_model.embed_query("Neurology?")
embedded_query[:5]

[0.023227831348776817,
 -0.06297047436237335,
 -0.023425033316016197,
 -0.00353750167414546,
 0.039694882929325104]

In [6]:
embedded_query = embeddings_model.embed_query("Dermatology?")
embedded_query[:5]

[0.041984859853982925,
 -0.06628748029470444,
 0.001272472902201116,
 -0.010246035642921925,
 0.0058945296332240105]